src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -90,6 +92,20 @@ static bool exclusive = false;
 #endif
 #define htonll(a) ntohll(a)
 
+static int do_unmap();
+
+static void handle_signal(int signum)
+{
+  assert(signum == SIGINT || signum == SIGTERM);
+  derr << "*** Got signal " << sig_str(signum) << " ***" << dendl;
+  dout(20) << __func__ << ": " << "sending NBD_DISCONNECT" << dendl;
+  if (ioctl(nbd, NBD_DISCONNECT) < 0) {
+    derr << "rbd-nbd: disconnect failed" << cpp_error(errno) << std::endl;
+  } else {
+    dout(20) << __func__ << ": " << "disconnected" << dendl;
+  }
```

Comment:
So the question is, why they are different? Does ```ioctl() < ``` means device is not used by all means? maybe we should check the ret-val, ENOENT means not used, but others means ```disconnect failed.```, right?